src/cms/utils/slug_utils.py

```diff
@@ -26,76 +63,72 @@ def generate_unique_slug(form_object, foreign_model):
+    :param kwargs: The supplied keyword arguments
+    :type kwargs: dict
 
     :raises ~django.core.exceptions.ValidationError: When no slug is given and there is also no field which can be used
                                                      as fallback (either ``title`` or ``name``).
 
     :return: An unique slug identifier
     :rtype: str
     """
+    slug = kwargs.get("slug")
+    foreign_model = kwargs.get("foreign_model")
+    object_instance = kwargs.get("object_instance")
+    fallback = kwargs.get("fallback")
+    cleaned_data = kwargs.get("cleaned_data")
+    region = kwargs.get("region")
+    language = kwargs.get("language")
 
     logger.info("generate_unique_slug()")
     logger.info('foreign_model: "%s"', foreign_model)
     if foreign_model in ["page", "event", "poi"]:
-        logger.info('region: "%s"', form_object.region)
-        logger.info('language: "%s"', form_object.language)
-
-    slug = form_object.cleaned_data["slug"]
-
-    # if slug is empty, generate from title/name
-    if not slug:
-        # determine fallback field of the model
-        if foreign_model in ["page", "event", "poi"]:
-            fallback = "title"
-        else:
-            fallback = "name"
-        if fallback not in form_object.cleaned_data:
+        logger.info('region: "%s"', region)
+        logger.info('language: "%s"', language)
+
+    # if slug is empty and fallback field is set, generate from fallback:title/name
+    if not slug and fallback:
+        if fallback not in cleaned_data:
             raise ValidationError(
                 _("Cannot generate slug from %(fallback)s."),
                 code="invalid",
                 params={"fallback": _(fallback)},
             )
         # slugify to make sure slug doesn't contain special chars etc.
-        slug = slugify(form_object.cleaned_data[fallback], allow_unicode=True)
+        slug = slugify(cleaned_data[fallback], allow_unicode=True)
         # If the title/name field didn't contain valid characters for a slug, we use a hardcoded fallback slug
         if not slug:
             slug = foreign_model
 
     unique_slug = slug
     i = 1
-    pre_filtered_objects = form_object.Meta.model.objects
+    pre_filtered_objects = kwargs["manager"]
 
     # if the foreign model is a content type (e.g. page, event or poi), make sure slug is unique per region and language
     if foreign_model in ["page", "event", "poi"]:
         pre_filtered_objects = pre_filtered_objects.filter(
             **{
-                foreign_model + "__region": form_object.region,
-                "language": form_object.language,
+                foreign_model + "__region": region,
+                "language": language,
             }
         )
 
     # generate new slug while it is not unique
     while True:
         # get other objects with same slug
         other_objects = pre_filtered_objects.filter(slug=unique_slug)
-        if form_object.instance.id:
+        if object_instance.id:
```

Comment:
```suggestion
        if object_instance and object_instance.id:
```
Sorry, I just noticed my suggestion fixes the form submission when the page doesn't exist, but the ajax call doesn't work in this case (it results in an error `AttributeError: 'NoneType' object has no attribute 'id'`)... so I think we have to check both - whether `object_instance` evaluates to `True` and whether it has an `id` already...

Reply:
No problem, when it comes to non existing page forms, I getting always into trouble too :sweat_smile: